src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -512,8 +512,7 @@ private void Resize()
 
         private void Resize(int newSize, bool forceNewHashCodes)
         {
-            Debug.Assert(newSize >= _entries.Length);
-
+            Debug.Assert(newSize >= Count);
```

Comment:
Why did you change this assert?

Reply:
I'll revert it. This won't need to change since we are not calling Resize on TrimExcess anymore